src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,360 @@
+package io.anserini.kg.freebase;
+
+import com.google.common.collect.MinMaxPriorityQueue;
+import io.anserini.kg.freebase.IndexTopics.TopicLuceneDocumentGenerator;
+import io.anserini.rerank.ScoredDocuments;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.core.SimpleAnalyzer;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.similarities.ClassicSimilarity;
+import org.apache.lucene.search.similarities.Similarity;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Lookups a Freebase mid and returns all properties associated with it.
+ */
+public class EntityLinking implements Closeable {
+  private static final Logger LOG = LogManager.getLogger(EntityLinking.class);
+
+  private final IndexReader reader;
+
+  private static int numHits = 200;
+
+  static final class Args {
+    // Required arguments
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-data", metaVar = "[file]", required =  true, usage = "dataset file")
+    public String data;
+
+    @Option(name = "-goldData", usage = "supplying with the ground truth dataset")
+    boolean goldData;
+
+    @Option(name = "-hits", metaVar = "hits", usage = "setting number of hits to be retrieved per query")
+    int hits;
+  }
+
+  class RankedEntity implements Comparable<RankedEntity> {
+    String mid;
+    float score;
+    String name;
+    String label;
+
+    RankedEntity(String m, float s, String n, String l) {
+      mid = m;
+      score = s;
+      name = n;
+      label = l;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("(%s, %f)", mid, score);
+    }
+
+    @Override
+    public int compareTo(RankedEntity o) {
+      return new Float(this.score).compareTo(new Float(o.score));
```

Comment:
I agree with Mina and would recommend using Float.compare() instead to avoid problems while comparing border values. Something like:
```java
public int compareTo(ScoredPassage o) {
    return Float.compare(score, o.score);
}
```